From f444bba95ac285d43f1c8125e3a83afc59f6b7c9 Mon Sep 17 00:00:00 2001
From: Ayyappa Ch <ayyappa.chandolu@amd.com>
Date: Wed, 15 Feb 2017 15:23:38 +0530
Subject: [PATCH 1717/1722] drm/amd/amdkfd : Eviction parameter should not
 updated when no queues are present

 Eviction of queues can be ignored if no queues are available.
 We should not decrement qpd->evicted if no queues are available
 amdgpu chnages creating fence eviction signal. This chnages is needed
 even we fix the actual reason from amdgpu for eviction

Signed-off-by: Ayyappa Ch <ayyappa.chandolu@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index 600d6c3..a5bf0c0 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -481,6 +481,16 @@ int process_evict_queues(struct device_queue_manager *dqm,
 	BUG_ON(!dqm || !qpd);
 
 	mutex_lock(&dqm->lock);
+	/* Eviction of queues can be ignored if no queues are available.
+	   We should not decrement qpd->evicted if no queues are available
+	   amdgpu chnages creating fence eviction signal. This chnages is needed
+	   even we fix the actual reason from amdgpu for eviction
+	*/
+	if(list_empty(&qpd->queues_list))  
+	{
+		mutex_unlock(&dqm->lock);
+		return 0;
+	}
 	if (qpd->evicted++ > 0) { /* already evicted, do nothing */
 		mutex_unlock(&dqm->lock);
 		return 0;
-- 
2.7.4

